src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java

```diff
@@ -5232,6 +5235,42 @@ public static Value can_faxbot(ScriptRuntime controller, final Value arg) {
     return DataTypes.FALSE_VALUE;
   }
 
+  public static Value can_faxbot(ScriptRuntime controller, final Value arg1, final Value arg2) {
+    MonsterData monster = (MonsterData) arg1.rawValue();
+    if (monster == null) {
+      return DataTypes.FALSE_VALUE;
+    }
+
+    String botName = arg2.toString().toLowerCase();
+
+    FaxBotDatabase.configure();
+
+    String actualName = monster.getName();
+    for (FaxBot bot : FaxBotDatabase.faxbots) {
+      if (bot == null) {
+        continue;
+      }
+
+      if (!botName.equals(bot.getName().toLowerCase())) {
+        continue;
+      }
+
+      Monster monsterObject = bot.getMonsterByActualName(actualName);
+      if (monsterObject == null) {
+        continue;
+      }
+
+      // Don't check if bot is online til we can do it safely for large numbers of can_faxbot
+      // requests
+      // if ( !FaxRequestFrame.isBotOnline( botName ) )
```

Comment:
I feel like this might have been added when minimizing server hits was more important than it is now: I think KoL's servers should easily be able to handle this.

Reply:
Darzil commented out the code in 2014 but the "Don't check..." was added by @heeheehee-kolmafia in 2021.

Reply:
I re-enabled it - and called the KoLmafia.isPlayerOnline primitive, rather than the FaxRequestFrame method which sets the status method in that frame - and get this now.
```
> ash can_faxbot($monster[knob goblin embezzler])

Configuring faxable monsters.
Configuring Easyfax (2504737)
Configuring CheeseFax (3038166)
Faxable monster lists fetched.
Returned: true

> ash can_faxbot($monster[knob goblin embezzler], "EasyFax")

Returned: true

> ash can_faxbot($monster[knob goblin embezzler], "CheeseFax")

Returned: false

> ash can_faxbot($monster[Black Crayon Constellation])

Returned: false

> ash can_faxbot($monster[Black Crayon Constellation], "EasyFax")

Returned: false

> ash can_faxbot($monster[Black Crayon Constellation], "CheeseFax")

Returned: false
```